src/Microsoft.Health.Fhir.SqlServer.UnitTests/Features/ChangeFeed/SqlServerFhirResourceChangeDataStoreTests.cs

```diff
@@ -75,5 +76,29 @@ public async Task GivenEmptyConnectionString_WhenGetResourceChanges_ThenInvalidO
                 Assert.Equal(nameof(InvalidOperationException), ex.GetType().Name);
             }
         }
+
+        [Fact]
+        public async Task GivenOperationCanceled_WhenGetResourceChanges_ThenNoExceptionThrown()
+        {
+            var mockConnection = new SqlConnectionStringBuilder
+            {
+                DataSource = "Test",
+                InitialCatalog = "Test",
+            }.ToString();
+
+            var config = Options.Create(new SqlServerDataStoreConfiguration { ConnectionString = mockConnection });
+            var connectionStringProvider = new DefaultSqlConnectionStringProvider(config);
+            var connectionFactory = new DefaultSqlConnectionFactory(connectionStringProvider);
+            var schemaInformation = new SchemaInformation(SchemaVersionConstants.Min, SchemaVersionConstants.Max);
+            schemaInformation.Current = SchemaVersionConstants.Max;
+            var resourceChangeDataStore = new SqlServerFhirResourceChangeDataStore(connectionFactory, NullLogger<SqlServerFhirResourceChangeDataStore>.Instance, schemaInformation);
+
+            CancellationTokenSource source = new CancellationTokenSource();
+            var token = source.Token;
+            source.Cancel();
+
+            var records = await resourceChangeDataStore.GetRecordsAsync(1, 1, token);
+            Assert.Equal(0, records.Count);
+        }
```

Comment:
I think we should check it with the actual exception type name like nameof(OperationCanceledException) or nameof(TaskCanceledException).

Reply:
updated to check for an exception now that we are throwing one